### PR TITLE
Integration with cuttlefish

### PR DESCRIPTION
**Pull request by joedevivo**
```
cuttlefish be cute as they are hard
```

Riak build with cuttlefish integration for config files! But what is cuttlefish? You'll see it soon enough!

### CODE REVIEW

**Comment by jaredmorrow** [comment]
Reminder a PR for EE will be needed as well.

**Comment by jaredmorrow** [comment]
Unable to build devrel for testing due to cuttlefish build error:

```
==> cuttlefish (compile)
Compiled src/conf_parse.peg
Compiled src/cuttlefish_vmargs.erl
/Volumes/Data/Development/basho/config/riak/deps/cuttlefish/src/lager_stderr_backend.erl:34: can't find include file "deps/lager/include/lager.hrl"
/Volumes/Data/Development/basho/config/riak/deps/cuttlefish/src/lager_stderr_backend.erl:61: undefined macro 'INT_LOG/3'
/Volumes/Data/Development/basho/config/riak/deps/cuttlefish/src/lager_stderr_backend.erl:24: function init/1 undefined
/Volumes/Data/Development/basho/config/riak/deps/cuttlefish/src/lager_stderr_backend.erl:115: function eol/0 is unused
/Volumes/Data/Development/basho/config/riak/deps/cuttlefish/src/lager_stderr_backend.erl:127: function is_new_style_console_available/0 is unused
ERROR: compile failed while processing /Volumes/Data/Development/basho/config/riak/deps/cuttlefish: rebar_abort
```

File: rel/files/riak.schema [hunk 1]
```diff
@@ -0,0 +1,1174 @@
...
+  {default, "{{node}}"}
+]}. 
+
+%% @doc Cookie for distributed node communication.  All nodes in the same cluster
+%% should use the same cookie or they will not be able to communicate.
+{mapping, "distributed_cookie", "vm_args.-setcookie", [
+  {default, "riak"}
+]}.
+
+{mapping, "erlang.asyc_threads", "vm_args.+A", [
```
Comment (seancribbs) [inline rel/files/riak.schema]: Typo: `async_threads`

**Comment by jaredmorrow** [comment]
On a fresh devrel build, no vm.args file is made so riak will not start:

```
➜  dev1 git:(jd-cuttlefish) ls etc
cert.pem  key.pem   riak.conf
➜  dev1 git:(jd-cuttlefish) ./bin/riak console
egrep: /Users/jared/Data/Development/basho/config/riak/dev/dev1/bin/../etc/vm.args: No such file or directory
vm.args needs to have either -name or -sname parameter.
```

**Comment by jaredmorrow** [comment]
Tested devrel, rel & package testing on CentOS 5.  +1 on this working with node_package's jd-cuttlefish branch.  I have not combed through the riak.schema, but I'm +1 from a release/packaging standpoint for Pre1.

Before this gets a final merge, I'd like to see it squashed, but I bet that was already planned.

**Comment by joedevivo** [comment]
@jaredmorrow there will be CSE review of the entire schema in the coming weeks.

**Comment by joedevivo** [comment]
going to close this, see #371

**Pull request closed without merging**